clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2760,9 +2760,10 @@ void handleFailure(Throwable throwable) {
             }, now);
         }
 
-        return new DescribeConfigsResult(new HashMap<>(nodeFutures.entrySet().stream()
-                .flatMap(x -> x.getValue().entrySet().stream())
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))));
+        return new DescribeConfigsResult(nodeFutures.entrySet()
```

Comment:
Please take a look at previous discussion: https://github.com/apache/kafka/pull/16250#issuecomment-2157809066

We need to make it create modifiable map explicitly.